File: solutions/elixir/lasagna/2/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven, do: 40

  def remaining_minutes_in_oven(m), do: expected_minutes_in_oven() - m

  def preparation_time_in_minutes(l), do: l * 2

  def total_time_in_minutes(l, m), do: preparation_time_in_minutes(l) + m

  def alarm, do: "Ding!"
end

File: solutions/swift/lasagna/1/Sources/Lasagna/Lasagna.swift
let expectedMinutesInOven = 40

func remainingMinutesInOven(elapsedMinutes: Int) -> Int { expectedMinutesInOven - elapsedMinutes }
func preparationTimeInMinutes(layers: Int) -> Int { 2 * layers }
func totalTimeInMinutes(layers: Int, elapsedMinutes: Int) -> Int {
  preparationTimeInMinutes(layers: layers) + elapsedMinutes
}
